src/webview/SampleView.ts

```diff
@@ -135,14 +136,34 @@ export class SampleView extends View {
     }
 
     private async getSamplePage(options: any): Promise<any> {
-
+        let validInternetConnection: boolean = true;
+        const outputAdapter: VSCodeBlockchainOutputAdapter = VSCodeBlockchainOutputAdapter.instance();
         const templatePath: string = path.join(__dirname, '..', '..', '..', 'templates', 'SampleView.ejs');
 
         const converter: showdown.Converter = new showdown.Converter();
-        const response: any = await Axios.get(options.sample.readme);
-        const text: string = response.data;
-        const html: string = converter.makeHtml(text);
-        options.html = html;
+        const repositoryRegistry: RepositoryRegistry = RepositoryRegistry.instance();
+        const repoIsCloned: boolean = repositoryRegistry.exists(options.repositoryName);
+        try {
+            const response: any = await Axios.get(options.sample.readme);
+            const text: string = response.data;
+            const html: string = converter.makeHtml(text);
+            options.html = html;
+        } catch (error) {
+            validInternetConnection = false;
+        }
+
+        if (!validInternetConnection) {
+            if (repoIsCloned) {
+                const clonedRepo: RepositoryRegistryEntry = repositoryRegistry.get(options.repositoryName);
+                const repoPath: string = clonedRepo.path;
+                const readMe: string = await fs.readFileSync(path.join(repoPath, 'README.md'), 'utf8');
```

Comment:
the readme might not always be at the top level, you need to get the readme property to work out where it is

Reply:
Fixed